pkg/endpoint/endpoint.go

```diff
@@ -1298,6 +1301,11 @@ func (e *Endpoint) SetStateLocked(toState, reason string) bool {
 		case StateDisconnecting, StateWaitingToRegenerate:
 			goto OKState
 		}
+	case StateRestoring:
+		switch toState {
+		case StateDisconnecting, StateRestoring, StateWaitingToRegenerate:
```

Comment:
When would we switch from StateRestoring to StateRestoring? I only see it as a start state (this might impact my healthz pr, I don't particularly care what the state transitions are).

Reply:
@raybejjani when we read the endpoints from the disk, we set the endpoint state with `StateRestoring` (`ep.state = StateRestoring`). later on, we can call to set state locked with
```
	ready := ep.SetStateLocked(endpoint.StateRestoring, "Triggering synchronous endpoint regeneration while syncing state to host")
```